### PR TITLE
[Openload] Fix extraction

### PR DESCRIPTION
**Pull request by sudovijay**
Just a minor fix for openload

## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [ ] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [ ] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [ ] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Explanation of your *pull request* in arbitrary form goes here. Please make sure the description explains the purpose and effect of your *pull request* and is worded well enough to be understood. Provide as much context and examples as possible.

### CODE REVIEW

File: youtube_dl/extractor/openload.py [hunk 1]
```diff
@@ -64,16 +64,17 @@ def _real_extract(self, url):
             raise ExtractorError('File not found', expected=True)
 
         ol_id = self._search_regex(
-            '<span[^>]+id="[a-zA-Z0-9]+x"[^>]*>([0-9]+)</span>',
+            '<span[^>]+?id="[^"]*?"[^>]*?>([^<]+?)<\/span>',
```
Comment (yan12125) [inline youtube_dl/extractor/openload.py]: 1. There's no need to escape slashes
2. Are question marks necessary?

Comment (sudovijay) [inline youtube_dl/extractor/openload.py]: i usually work in php so i'm used to of escaping slashes that ain't required ! nor the question marks

File: youtube_dl/extractor/openload.py [hunk 2]
```diff
@@ -64,16 +64,17 @@ def _real_extract(self, url):
...
             webpage, 'openload ID')
 
-        first_two_chars = int(float(ol_id[0:][:2]))
+        first_three_chars = int(float(ol_id[0:][:3]))
```
Comment (adogyf) [inline youtube_dl/extractor/openload.py]: first_three_chars = int(float(ol_id[0:3])) is cleaner

File: youtube_dl/extractor/openload.py [hunk 3]
```diff
@@ -64,16 +64,17 @@ def _real_extract(self, url):
             raise ExtractorError('File not found', expected=True)
 
         ol_id = self._search_regex(
-            '<span[^>]+id="[a-zA-Z0-9]+x"[^>]*>([0-9]+)</span>',
+            '<span[^>]+id="[^"]*"[^>]*>([^<]+)</span>',
```
Comment (yan12125) [inline youtube_dl/extractor/openload.py]: Should be [^"]+ instead of [^"]*

File: youtube_dl/extractor/openload.py (hunk 2 above)
Comment (adogyf) [inline youtube_dl/extractor/openload.py]: first_three_chars = int(float(ol_id[0:3])) is cleaner

File: youtube_dl/extractor/openload.py (hunk 3 above)
Comment (mdoulabi) [inline youtube_dl/extractor/openload.py]: i cant find this youtube_dl , what is patch address?

**Comment by dstftw** [comment]
Cherry-picked, thanks.

**Comment by yan12125** [comment]
@mdoulabi Openload is already fixed in the latest version. See https://github.com/rg3/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl

**Comment by mdoulabi** [comment]
i use youtube-dl -U and update 
2017.01.14
but cant download openload.
why?

**Comment by yan12125** [comment]
Works for me. Please paste your log.
```
$ youtube-dl -v "https://openload.co/f/kUEfGclsU9o"
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-v', 'https://openload.co/f/kUEfGclsU9o']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2017.01.14
[debug] Git HEAD: b80e2ebc8
[debug] Python version 3.6.0 - Linux-4.9.0-1-ARCH-x86_64-with-arch-Arch-Linux
[debug] exe versions: ffmpeg 3.2.2, ffprobe 3.2.2, rtmpdump 2.4
[debug] Proxy map: {}
[Openload] kUEfGclsU9o: Downloading webpage
[debug] Invoking downloader on 'https://openload.co/stream/kUEfGclsU9o~1484565089~140.112.0.0~qITYEysW'
[download] Destination: skyrim_no-audio_1080.mp4-kUEfGclsU9o.mp4
[download] 100% of 9.80MiB in 00:13
```

**Comment by mdoulabi** [comment]
[root@server ~]# youtube-dl -v "https://openload.co/f/kUEfGclsU9o"
[debug] System config: []
[debug] User config: [u'-f', u'bestvideo[height<=720]+bestaudio/best[height<=720                                                                                        ]']
[debug] Custom config: []
[debug] Command-line args: [u'-v', u'https://openload.co/f/kUEfGclsU9o']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2017.01.14
[debug] Python version 2.6.6 - Linux-2.6.32-573.el6.x86_64-x86_64-with-centos-6.                                                                                        7-Final
[debug] exe versions: ffmpeg N-78980-ga008a7c, ffprobe 2.2.1
[debug] Proxy map: {}
[Openload] kUEfGclsU9o: Downloading webpage
ERROR: requested format not available
Traceback (most recent call last):
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 705, in extract                                                                                        _info
    return self.process_ie_result(ie_result, download, extra_info)
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 751, in process                                                                                        _ie_result
    return self.process_video_result(ie_result, download=download)
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 1427, in proces                                                                                        s_video_result
    expected=True)
ExtractorError: requested format not available

**Comment by yan12125** [comment]
There's no format satisfying ```bestvideo[height<=720]+bestaudio/best[height<=720 ]```

**Comment by mdoulabi** [comment]
how should chnage this  bestvideo[height<=720]+bestaudio/best[height<=720 ]?

**Comment by yan12125** [comment]
bestvideo[height<=720]+bestaudio/best[height<=720 ]/best

**Comment by mdoulabi** [comment]
this line : bestvideo[height<=720]+bestaudio/best[height<=720 ]/best .
i should which file?

**Comment by yan12125** [comment]
Depends. See https://github.com/rg3/youtube-dl/blob/master/README.md#configuration

**Comment by mdoulabi** [comment]
ow . its ok  i use of --ignore-config  parameter

**Pull request closed without merging**